File: tools/pas2php-code-6/pas2php_include/__variants.pas
{
 ***************************************************************************
 *                                                                         *
 *   This source is free software; you can redistribute it and/or modify   *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 *   This code is distributed in the hope that it will be useful, but      *
 *   WITHOUT ANY WARRANTY; without even the implied warranty of            *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU     *
 *   General Public License for more details.                              *
 *                                                                         *
 *   A copy of the GNU General Public License is available on the World    *
 *   Wide Web at <http://www.gnu.org/copyleft/gpl.html>. You can also      *
 *   obtain it by writing to the Free Software Foundation,                 *
 *   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.        *
 *                                                                         *
 ***************************************************************************
}

unit __Variants;

interface

uses PhpLib, SysUtils;

function VarArrayHighBound(const AVariant: variant; const ADim: integer): integer;
function VarArrayLowBound(const AVariant: variant; const ADim: integer): integer;
function VarArrayOf(const AVariant: variant): variant;
function VarIsArray(const AVariant: variant): boolean;
function VarIsBool(const AVariant: variant): boolean;
function VarIsFloat(const AVariant: variant): boolean;
function VarIsNull(const AVariant: variant): boolean;
function VarIsNumeric(const AVariant: variant): boolean;
function VarIsStr(const AVariant: variant): boolean;
function VarToStr(const AVariant: variant): string;
function VarToStrDef(const AVariant: variant; const ADefault: string): string;
procedure VarArrayRedim(var AVariant: variant; const AHigh: integer);

implementation

function VarIsArray(const AVariant: variant): boolean;
begin
  Result := php_is_array(AVariant);
end;

function VarIsBool(const AVariant: variant): boolean;
begin
  Result := php_is_bool(AVariant);
end;

function VarIsFloat(const AVariant: variant): boolean;
begin
  Result := php_is_float(AVariant);
end;

function VarIsNull(const AVariant: variant): boolean;
begin
  Result := php_is_null(AVariant);
end;

function VarIsNumeric(const AVariant: variant): boolean;
begin
  Result := php_is_numeric(AVariant);
end;

function VarIsStr(const AVariant: variant): boolean;
begin
  Result := php_is_string(AVariant);
end;

function VarToStrDef(const AVariant: variant; const ADefault: string): string;
begin
  if VarIsNull(AVariant) then begin
    Result := ADefault;
  end else begin
    Result := php_strval(AVariant);
  end;
end;

function VarToStr(const AVariant: variant): string;
begin
  Result := VarToStrDef(AVariant, EmptyStr);
end;

function VarArrayOf(const AVariant: variant): variant;
begin
  Result := AVariant;
end;

function VarArrayLowBound(const AVariant: variant; const ADim: integer): integer;
begin
  Assert(ADim = 1);
  Result := 0;
end;

function VarArrayHighBound(const AVariant: variant; const ADim: integer): integer;
begin
  Assert(ADim = 1);
  Result := php_count(AVariant) - 1;
end;

procedure VarArrayRedim(var AVariant: variant; const AHigh: integer);
var
  LIndex: integer;
begin
  for LIndex := php_count(AVariant) - 1 downto AHigh + 1 do begin
    php_unset(AVariant[LIndex]);
  end;
  for LIndex := php_count(AVariant) to AHigh do begin
    AVariant[LIndex] := Null;
  end;
end;

end.

File: tools/pas2php-code-6/pas2php_include/variants.php
<?PHP
// 
// Converted From Pascal By Pas2Php v0.8
// Converts Object Pascal to PHP
// http://www.wascal.net/?page=pas2php
// Pas2Php v0.8 is Copyright (C) 2014-2016 Derek John Evans
// 
include_once('system.php');
include_once('phplib.php');
include_once('sysutils.php');
// function VarIsArray(const AVariant: variant) : Boolean;
function VarIsArray($AVariant) {
  // Result : Boolean;
  $Result = false;
  $Result = is_array($AVariant);
  return $Result;
}
// function VarIsBool(const AVariant: variant) : Boolean;
function VarIsBool($AVariant) {
  // Result : Boolean;
  $Result = false;
  $Result = is_bool($AVariant);
  return $Result;
}
// function VarIsFloat(const AVariant: variant) : Boolean;
function VarIsFloat($AVariant) {
  // Result : Boolean;
  $Result = false;
  $Result = is_float($AVariant);
  return $Result;
}
// function VarIsNull(const AVariant: variant) : Boolean;
function VarIsNull($AVariant) {
  // Result : Boolean;
  $Result = false;
  $Result = is_null($AVariant);
  return $Result;
}
// function VarIsNumeric(const AVariant: variant) : Boolean;
function VarIsNumeric($AVariant) {
  // Result : Boolean;
  $Result = false;
  $Result = is_numeric($AVariant);
  return $Result;
}
// function VarIsStr(const AVariant: variant) : Boolean;
function VarIsStr($AVariant) {
  // Result : Boolean;
  $Result = false;
  $Result = is_string($AVariant);
  return $Result;
}
/*
function VarToStrDef(const AVariant: variant;const ADefault: string)
                     : string;
*/
function VarToStrDef($AVariant, $ADefault) {
  // Result : string;
  $Result = "";
  if (VarIsNull($AVariant)) {
    $Result = $ADefault;
  }
  else {
    $Result = strval($AVariant);
  }
  return $Result;
}
// function VarToStr(const AVariant: variant) : string;
function VarToStr($AVariant) {
  // Result : string;
  $Result = "";
  $Result = VarToStrDef($AVariant, EmptyStr);
  return $Result;
}
// function VarArrayOf(const AVariant: variant) : variant;
function VarArrayOf($AVariant) {
  // Result : variant;
  $Result = null;
  $Result = $AVariant;
  return $Result;
}
/*
function VarArrayLowBound(const AVariant: variant;const ADim: Integer)
                          : Integer;
*/
function VarArrayLowBound($AVariant, $ADim) {
  // Result : Integer;
  $Result = 0;
  Enforce($ADim == 1);
  $Result = 0;
  return $Result;
}
/*
function VarArrayHighBound(const AVariant: variant;const ADim: Integer)
                           : Integer;
*/
function VarArrayHighBound($AVariant, $ADim) {
  // Result : Integer;
  $Result = 0;
  Enforce($ADim == 1);
  $Result = count($AVariant) - 1;
  return $Result;
}
// procedure VarArrayRedim(var AVariant: variant;const AHigh: Integer);
function VarArrayRedim(&$AVariant, $AHigh) {
  // LIndex : Integer;
  $LIndex = 0;
  for ($LIndex = count($AVariant) - 1, $LIndex_High_ = $AHigh + 1; $LIndex >= $LIndex_High_; $LIndex--) {
    unset($AVariant[$LIndex]);
  }
  for ($LIndex = count($AVariant), $LIndex_High_ = $AHigh; $LIndex <= $LIndex_High_; $LIndex++) {
    $AVariant[$LIndex] = Null;
  }
}
?>